datalad/distribution/create.py

```diff
@@ -157,61 +162,69 @@ def __call__(
-                annex_version=annex_version,
-                annex_backend=annex_backend,
-                git_opts=git_opts,
-                annex_opts=annex_opts,
-                annex_init_opts=annex_init_opts)
 
-        else:
-            if no_annex:
-                if description:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "description for annex repo and declaring "
-                                     "no annex repo.")
-                if annex_opts:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "options for annex and declaring no "
-                                     "annex repo.")
-                if annex_init_opts:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "options for annex init and declaring no "
-                                     "annex repo.")
-
-                lgr.info("Creating a new git repo at %s", ds.path)
-                vcs = GitRepo(ds.path, url=None, create=True,
-                              git_opts=git_opts)
+            if sds is not None:
+                return sds.create_subdataset(
+                    ds.path,
+                    name=name,
+                    description=description,
+                    no_annex=no_annex,
+                    annex_version=annex_version,
+                    annex_backend=annex_backend,
+                    git_opts=git_opts,
+                    annex_opts=annex_opts,
+                    annex_init_opts=annex_init_opts)
             else:
-                # always come with annex when created from scratch
-                lgr.info("Creating a new annex repo at %s", ds.path)
-                vcs = AnnexRepo(ds.path, url=None, create=True,
-                                backend=annex_backend,
-                                version=annex_version,
-                                description=description,
-                                git_opts=git_opts,
-                                annex_opts=annex_opts,
-                                annex_init_opts=annex_init_opts)
-                # record the annex uuid of this repo for this afterlife
-                # to be able to track siblings and children
-                if 'datalad.annex.origin' in ds.config:
-                    # make sure we reset this variable completely, in case of a re-create
-                    ds.config.unset('datalad.annex.origin', where='dataset')
-                ds.config.add('datalad.annex.origin', vcs.uuid, where='dataset')
-                ds.repo.add('.datalad', git=True)
-
-            vcs.commit(msg="datalad initial commit",
-                       options=to_options(allow_empty=True))
-            # reset ID, we have a VCS now
-            ds._id = None
-            return ds
+                if isinstance(add_to_super, bool):
```

Comment:
On Fri, 02 Sep 2016, Benjamin Poldrack wrote:

>    In datalad/distribution/create.py:

>  > -                                annex_opts=annex_opts,
>  > -                                annex_init_opts=annex_init_opts)
>  > -                # record the annex uuid of this repo for this afterlife
>  > -                # to be able to track siblings and children
>  > -                if 'datalad.annex.origin' in ds.config:
>  > -                    # make sure we reset this variable completely, in case of a re-create
>  > -                    ds.config.unset('datalad.annex.origin', where='dataset')
>  > -                ds.config.add('datalad.annex.origin', vcs.uuid, where='dataset')
>  > -                ds.repo.add('.datalad', git=True)
>  > -
>  > -            vcs.commit(msg="datalad initial commit",
>  > -                       options=to_options(allow_empty=True))
>  > -            # reset ID, we have a VCS now
>  > -            ds._id = None
>  > -            return ds
>  > +                if isinstance(add_to_super, bool):

>    Well, how about `if add_to_super is True` instead of just `if
>    add_to_super`? ;-) Anyways - do as you like it. 

I kinda trust None to be None, but somehow distract True to be the same
True we think about (at least not in python 2):

$> cat true.py
True = 0
print("True==%r" % True)

$> python-all true.py
Testing using /usr/bin/python2.7
True==0
Testing using /usr/bin/python3.4
  File "true.py", line 1
    True = 0
    ^
SyntaxError: can't assign to keyword
Testing using /usr/bin/python3.5
  File "true.py", line 1
    True = 0
    ^
SyntaxError: can't assign to keyword
Testing using /usr/bin/python2.7-dbg
True==0
[21664 refs]
Testing using /usr/bin/python3.4-dbg
  File "true.py", line 1
    True = 0
    ^
SyntaxError: can't assign to keyword
Testing using /usr/bin/python3.5-dbg
  File "true.py", line 1
    True = 0
    ^
SyntaxError: can't assign to keyword
Testing using /usr/local/bin/python3.3
  File "true.py", line 1
    True = 0
    ^
SyntaxError: can't assign to keyword


$> cat none.py       
None = "not none"

$> python-all none.py 
Testing using /usr/bin/python2.7
  File "none.py", line 1
    None = "not none"
SyntaxError: cannot assign to None
Testing using /usr/bin/python3.4
  File "none.py", line 1
    None = "not none"
    ^
SyntaxError: can't assign to keyword
Testing using /usr/bin/python3.5
  File "none.py", line 1
    None = "not none"
    ^
SyntaxError: can't assign to keyword
Testing using /usr/bin/python2.7-dbg
  File "none.py", line 1
    None = "not none"
SyntaxError: cannot assign to None
[21664 refs]
Testing using /usr/bin/python3.4-dbg
  File "none.py", line 1
    None = "not none"
    ^
SyntaxError: can't assign to keyword
Testing using /usr/bin/python3.5-dbg
  File "none.py", line 1
    None = "not none"
    ^
SyntaxError: can't assign to keyword
Testing using /usr/local/bin/python3.3
  File "none.py", line 1
    None = "not none"
    ^
SyntaxError: can't assign to keyword